components/operator/internal/resources/ledgers/deployments.go

```diff
@@ -27,17 +27,34 @@ import (
 func installLedger(ctx core.Context, stack *v1beta1.Stack,
 	ledger *v1beta1.Ledger, database *v1beta1.Database, image string, isV2 bool) error {
 
+	replicasSettings, err := settings.GetStringOrDefault(ctx, stack.Name, "1", "deployments", "ledger", "replicas")
+	if err != nil {
+		return err
+	}
+
 	switch ledger.Spec.DeploymentStrategy {
 	case v1beta1.DeploymentStrategyMonoWriterMultipleReader:
 		if err := core.DeleteIfExists[*appsv1.Deployment](ctx, core.GetNamespacedResourceName(stack.Name, "ledger")); err != nil {
 			return err
 		}
 		return installLedgerMonoWriterMultipleReader(ctx, stack, ledger, database, image, isV2)
-	default:
+	case v1beta1.DeploymentStrategySingle:
 		if err := uninstallLedgerMonoWriterMultipleReader(ctx, stack); err != nil {
 			return err
 		}
 		return installLedgerSingleInstance(ctx, stack, ledger, database, image, isV2)
+	default:
```

Comment:
Ugly copy/paste :D

You could have define a variable `isSingle` before the switch to determine the mode following either the crd property or the setting.
Then replace the switch by a if/else.